### PR TITLE
Make Saturation Function Family Available in Runspec

### PR DESCRIPTION
**Pull request by bska**
This information is generally useful so we should have a single source of truth and not have to reimplement the logic every time we need it.

### CODE REVIEW

**Comment by bska** [comment]
I will create a follow-up PR to switch the [`SatFuncPropertyInitializers`]() to use this public version of the information.  For now, I mainly need this to implement `KRWR`.

**Comment by bska** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Runspec.cpp [hunk 1]
```diff
@@ -16,19 +16,110 @@
...
+        // solvent simulations.
+
+        const auto family2 =      // SGFN, SOF{2,3}, SWFN
+            (gas && deck.hasKeyword<Opm::ParserKeywords::SGFN>()) ||
+            (oil && ((threeP && deck.hasKeyword<Opm::ParserKeywords::SOF3>()) ||
+                     (twoP && deck.hasKeyword<Opm::ParserKeywords::SOF2>()))) ||
+            (wat && deck.hasKeyword<Opm::ParserKeywords::SWFN>());
+
+        if (gas &&
+            deck.hasKeyword<Opm::ParserKeywords::SGOF>() &&
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Runspec.cpp:99]: In situations like this - throwing a fatal exception based on invalid deck input I try to output the keywordlocation; i.e. something like this:
```C+++
{
     const auto& sgof_location = deck.getKeyword<Opm::ParserKeywords::SGOF>().location();
     const auto& slgof_location = deck.getKeyword<Opm::ParserKeywords::SLGOF>().location();

     msg = fmt::format("Tables SGOF and SLGOF are mutually exclusive\n"
                        "SGOF: line {} in {}\n"
                        "SLGOF: line {} in {}", sgof_location.lineno, sgof_location.filename, 
                                                slgof_location.lineno, slgof_location.filename);
    throw std::invalid_argument(msg);
``` 
}

Comment (bska) [reply src/opm/parser/eclipse/EclipseState/Runspec.cpp:99]: > throwing a fatal exception based on invalid deck input I try to output the keyword location;

We may even add `prohibits` clauses to the keywords to ensure we never get here.  Do those emit location information? If not I'll do what you suggest.

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Runspec.cpp:99]: > We may even add prohibits clauses to the keywords to ensure we never get here.

yes - that is probably the best solution.

> Do those emit location information? 

Yes [they do](https://github.com/OPM/opm-common/blob/master/src/opm/parser/eclipse/Parser/Parser.cpp#L590)

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Pull request merged**